Use Guard-Malloc modifier for webgl/2.0.0/conformance2/textures/misc/tex-unpack-params.html

rniwa opened the pull request:
#### 880116062398261bb7bab99b014c5ee3a830be95
<pre>
Use Guard-Malloc modifier for webgl/2.0.0/conformance2/textures/misc/tex-unpack-params.html
<a href="https://bugs.webkit.org/show_bug.cgi?id=250435">https://bugs.webkit.org/show_bug.cgi?id=250435</a>

Reviewed by Alexey Proskuryakov.

* LayoutTests/platform/mac/TestExpectations:

Canonical link: <a href="https://commits.webkit.org/259008@main">https://commits.webkit.org/259008@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8ed5295e07a62dd635d847e2a9ac6de8c31200e3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/103600 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/12715 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/36553 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112833 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/173134 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/13741 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/3614 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95851 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/111999 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/109372 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/10585 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/93646 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38317 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/92409 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25249 "Found 1 new test failure: http/tests/media/fairplay/fps-hls-key-rotation.html (failure)") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/79976 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6091 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26650 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6272 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3179 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12253 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46161 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8024 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3290 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on LayoutTests/platform/mac/TestExpectations:
```diff
@@ -1743,13 +1743,13 @@ inspector/unit-tests/test-harness-expect-functions.html [ Pass Crash ]
 
 # <rdar://problem/60873976> [ Mac Guard Malloc ] imported/w3c/web-platform-tests/WebCryptoAPI/generateKey/failures_AES-CBC.https.any.worker.html is flaky crashing.
-imported/w3c/web-platform-tests/WebCryptoAPI/generateKey/failures_AES-CBC.https.any.worker.html [ Pass Crash ]
+[ Guard-Malloc ] imported/w3c/web-platform-tests/WebCryptoAPI/generateKey/failures_AES-CBC.https.any.worker.html [ Pass Crash ]
```
It's almost never the case that a flaky crash in GuardMalloc only. If there is memory corruption, it's not GuardMalloc specific, and we should probably be skipping such tests outright to reduce downstream effects.

I do see flaky crashes on this one in release.

aproskuryakov commented on LayoutTests/platform/mac/TestExpectations:
```diff
@@ -1743,13 +1743,13 @@ inspector/unit-tests/test-harness-expect-functions.html [ Pass Crash ]
 
 # <rdar://problem/61264058> [ iOS and Mac with Guard-Malloc and wk2 or Stress-GC ] imported/w3c/web-platform-tests/css/css-sizing/percentage-height-in-flexbox.html is flaky failing.
-imported/w3c/web-platform-tests/css/css-sizing/percentage-height-in-flexbox.html [ Pass Failure ]
+[ Guard-Malloc ] imported/w3c/web-platform-tests/css/css-sizing/percentage-height-in-flexbox.html [ Pass Failure ]
```
Same thing - what is GuardMalloc specific here?

I think that we should only use this expectation for tests that are somehow invalid underGuardMalloc (the only case I know of is allocating too much memory).

aproskuryakov commented on LayoutTests/platform/mac/TestExpectations:
```diff
@@ -1798,7 +1798,7 @@ webkit.org/b/209692 [ BigSur+ ] platform/mac/fast/text/text-security-disc-bullet
 
 # rdar://63094384 (REGRESSION: (r261413): [ Guard-Malloc ] webgl/2.0.0/conformance2/textures/misc/tex-unpack-params.html is crashing consistently with - libGLImage.dylib: void glgConvertTo_32<GLGConverter_RGB8_ARGB8, (GLGMemory)1>)
-[ BigSur+ ] webgl/2.0.0/conformance2/textures/misc/tex-unpack-params.html [ Skip ]
+[ BigSur+ Guard-Malloc ] webgl/2.0.0/conformance2/textures/misc/tex-unpack-params.html [ Skip ]
```
Looking at the bug, it's weird, and I'm not sure if we even need to skip this under GuardMalloc any more. This does look like a step in the right direction of sure.

Nit:  BigSur+ is not needed any more.

rniwa committed:
Use Guard-Malloc modifier for webgl/2.0.0/conformance2/textures/misc/tex-unpack-params.html
https://bugs.webkit.org/show_bug.cgi?id=250435

Reviewed by Alexey Proskuryakov.

* LayoutTests/platform/mac/TestExpectations:

Canonical link: https://commits.webkit.org/259008@main
